hazelcast/src/main/java/com/hazelcast/config/PersistentMemoryDirectoryConfig.java

```diff
@@ -0,0 +1,142 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+import java.util.Objects;
+
+/**
+ * Configuration class for directories that persistent memories are mounted at.
+ */
+public class PersistentMemoryDirectoryConfig {
+    private static final int NUMA_NODE_UNSET = -1;
+
+    private String directory;
+    private int numaNode = NUMA_NODE_UNSET;
+
+    /**
+     * Creates an instance with the {@code directory} specified.
+     *
+     * @param directory The directory where the persistent memory is
+     *                  mounted at
+     */
+    public PersistentMemoryDirectoryConfig(String directory) {
+        this.directory = directory;
+    }
+
+    /**
+     * Creates an instance with the {@code directory} and NUMA node specified.
+     *
+     * @param directory The directory where the persistent memory is
+     *                  mounted at
+     * @param numaNode  The NUMA node that the persistent memory mounted
+     *                  to the given directory is attached to
+     */
+    public PersistentMemoryDirectoryConfig(String directory, int numaNode) {
+        this.directory = directory;
+        this.numaNode = numaNode;
+    }
+
+    /**
+     * Constructs an instance by copying the provided {@link PersistentMemoryDirectoryConfig}.
+     *
+     * @param directoryConfig The configuration to copy
+     */
+    public PersistentMemoryDirectoryConfig(PersistentMemoryDirectoryConfig directoryConfig) {
+        this.directory = directoryConfig.directory;
+        this.numaNode = directoryConfig.numaNode;
+    }
+
+    /**
+     * Returns the directory of this {@link PersistentMemoryDirectoryConfig}.
+     *
+     * @return the directory
+     */
+    public String getDirectory() {
+        return directory;
+    }
+
+    /**
+     * Returns the {@code directory} of this {@link PersistentMemoryDirectoryConfig}.
+     *
+     * @param directory the directory to set
+     */
+    public void setDirectory(String directory) {
+        this.directory = directory;
+    }
+
+    /**
+     * Returns the NUMA node the persistent memory mounted to the given
+     * directory is attached to.
+     *
+     * @return the NUMA node of the persistent memory
+     */
+    public int getNumaNode() {
+        return numaNode;
+    }
+
+    /**
+     * Sets the NUMA node the persistent memory mounted to the given
+     * directory is attached to.
+     *
+     * @param numaNode the NUMA node to set
```

Comment:
Additionally, here and in the constructor you can write that setting it to -1 disables it.

Reply:
:+1: Done.